thread_sync.c

```diff
@@ -7,7 +7,7 @@ static VALUE rb_eClosedQueueError;
 
 /* Mutex */
 typedef struct rb_mutex_struct {
-    rb_fiber_t *fiber;
+    rb_serial_t fiber_serial;
```

Comment:
Sorry if this is a stupid question, but if we are just considering the opaque memory address of the fiber, isn't that semantically equivalent to a serial number?

I guess the only limitation might be if it's getting reused?

Reply:
Yep! That was the previous issue fixed in https://bugs.ruby-lang.org/issues/21342